internal/app/singularity/cache_list_linux.go

```diff
@@ -0,0 +1,207 @@
+			if err != nil {
+				return fmt.Errorf("Unable to get stat: %v", err)
+			}
+			printFileSize, err := findSize(fileInfo.Size())
+			if err != nil {
+				// no need to describe the error, since it is already
+				sylog.Warningf("%v", err)
+			}
+			fmt.Printf("%-22s %-22s %-16s %s\n", b.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "oci")
+		}
+	}
+	return nil
+}
+
+func listBlobCache(printList bool) error {
+	// loop thrught ociBlob cache
+	count := 0
+	var totalSize int64
+
+	_, err = os.Stat(filepath.Join(cache.OciBlob(), "/blobs"))
+	if os.IsNotExist(err) {
+		return nil
+	}
+	blobs, err := ioutil.ReadDir(filepath.Join(cache.OciBlob(), "/blobs/"))
+	if err != nil {
+		return fmt.Errorf("Unable to opening oci folder: %v", err)
+	}
+	for _, f := range blobs {
+		blob, err := ioutil.ReadDir(filepath.Join(cache.OciBlob(), "/blobs/", f.Name()))
+		if err != nil {
+			return fmt.Errorf("Unable to looking in cache: %v", err)
+		}
+		for _, b := range blob {
+			fileInfo, err := os.Stat(filepath.Join(cache.OciBlob(), "/blobs/", f.Name(), b.Name()))
+			if err != nil {
+				return fmt.Errorf("Unable to get stat: %v", err)
+			}
+			if printList == true {
+				printFileSize, err := findSize(fileInfo.Size())
+				if err != nil {
+					// no need to describe the error, since it is already
+					sylog.Warningf("%v", err)
+				}
+				fmt.Printf("%-22.20s %-22s %-16s %s\n", b.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "blob")
+			}
+			count++
+			totalSize += fileInfo.Size()
+		}
+	}
+	if printList != true && count >= 1 {
+		printFileSize, err := findSize(totalSize)
+		if err != nil {
+			// no need to describe the error, since it is already
+			sylog.Warningf("%v", err)
+		}
+		fmt.Printf("\nThere are %d oci blob file(s) using %v of space. Use: '-T=blob' to list\n", count, printFileSize)
+	}
+	return nil
+}
+
+// ListSingularityCache : list local singularity cache, typeNameList : is a string of what cache
+// to list (seprate each type with a comma; like this: library,oci,blob) allList : force list all cache.
+func ListSingularityCache(cacheListTypes []string, listAll bool) error {
+	libraryList := false
+	ociList := false
+	blobList := false
+	listBlobSum := false
+
+	// split the string for each `,` then loop throught it and find what flags are there.
+	// then see whats true/false later. heres the benefit of doing it like this; if the user
+	// specified `library` twice, it will still only be printed once.
+	if len(cacheListTypes) >= 1 {
+		for _, t := range cacheListTypes {
+			switch t {
+			case "library":
+				libraryList = true
+			case "oci":
+				ociList = true
+			case "blob", "blobs":
+				blobList = true
+			case "blobSum":
+				listBlobSum = true
+			case "all":
+				listAll = true
+			default:
+				sylog.Fatalf("Not a valid type: %v", t)
+				os.Exit(2)
+			}
+		}
+	}
+
+	fmt.Printf("%-22s %-22s %-16s %s\n", "NAME", "DATE CREATED", "SIZE", "TYPE")
+
+	if listAll == true {
```

Comment:
I think we need to remove a bit of the complexity from the code here. There's no reason we need to have multiple calls here. I would propose to delete the entire `if listAll {...}` block. We can handle the `listAll` case in the next set of conditionals.

Reply:
`listAll` Is suppose to list all cache types (library, oci, & blob), and also override any other option.
I'm not sure how `if libraryList || listAll {...}` will list all cache if `listAll` is `true`... Maybe I'm missing something...

Reply:
Wait! nevermined... I get it now, (sorry I'm a little slow sometimes)